src/mono/mono/mini/interp/transform.c

```diff
@@ -2504,6 +2504,19 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 		// and always return false for IsDynamicCodeCompiled
 		if (!strcmp (tm, "get_IsDynamicCodeCompiled"))
 			*op = MINT_LDC_I4_0;
+	} else if (in_corlib &&
+			!strncmp ("System.Runtime.Intrinsics.X86", klass_name_space, 29) &&
+			(!strcmp (klass_name, "Lzcnt") ||
+			(m_class_get_nested_in (target_method->klass) &&
+			!strcmp (m_class_get_name (m_class_get_nested_in (target_method->klass)), "Lzcnt")))) {
```

Comment:
@tannergooding Note that I'm intrinsifying `Lzcnt` on all arches, even though this is technically a X86 only intrinsic. Do you think this is fine ?

Reply:
@BrzVlad given that the wasm simd opts were basically picked as a subset with broad arch support would it make sense to focus on those?  I guess there is the obvious downside at the moment that many code paths don't enable them at the moment.

cc @radekdoulik

Reply:
> Do you think this is fine ?

I'd expect it to cause downstream failures. We have explicit tests that `Lzcnt.LeadingZeroCount` throw if `Lzcnt.IsSupported` returns `false` and that it should be returning `false` on non xarch based platforms.

Other platforms, such as `Arm64` and `Wasm`, have their own intrinsics to use (`ArmBase.LeadingZeroCount` or `WasmBase.LeadingZeroCount`).

There are then the cross platform abstractions exposed via `System.Numerics.BitOperations` (e.g. `BitOperations.LeadingZeroCount`) and the generic math interfaces (e.g. `System.Int32.LeadingZeroCount`) which internally check each of the platform specific ISAs and provide a software fallback.

RyuJIT currently recognizes and treats both platform specific and the higher level abstractions as intrinsic to help with JIT throughput and overall codegen.

Reply:
I will instead intrinsify `Lzcnt.LeadingZeroCount`, `ArmBase.LeadingZeroCount` or `WasmBase.LeadingZeroCount` depending on the architecture that interp runs on.